### PR TITLE
wifi: eswifi: cleanups related to using devicetree constructs

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

**Comment by galak** [comment]
Do we need `csn` on the spi bus side?  I don't see it used anywhere.

**Comment by erwango** [comment]
> > Do we need `csn` on the spi bus side? I don't see it used anywhere.
> 
> It seems to be a chip select duplicated entry. I think you should delete it. FYI: @erwango

Agree seems like a leftover of a previous cleanup

**Comment by galak** [comment]
@rlubos can you take a look again at this PR.

**Review by rlubos — APPROVED** [review]

File: drivers/wifi/eswifi/eswifi_bus_spi.c [hunk 1]
```diff
@@ -22,13 +22,21 @@ K_KERNEL_STACK_MEMBER(eswifi_spi_poll_stack, ESWIFI_SPI_THREAD_STACK_SIZE);
 
 #define SPI_READ_CHUNK_SIZE 32
 
+struct eswifi_spi_config {
+	struct gpio_dt_spec csn;
```
Comment (gmarull) [inline drivers/wifi/eswifi/eswifi_bus_spi.c:26]: this one is not initialized?

Comment (galak) [reply drivers/wifi/eswifi/eswifi_bus_spi.c:26]: was going to send a follow up PR to remove that after this got merged.

**Review by gmarull — APPROVED** [review]

**Pull request merged**